xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/StringClass.java

```diff
@@ -123,4 +124,27 @@ public void displayEdit(StringBuffer buffer, String name, String prefix, BaseCol
         buffer.append(input.toString());
     }
 
+    @Override
+    public void displayView(StringBuffer buffer, String name, String prefix, BaseCollection object,
+        XWikiContext context)
+    {
+        String content = getPropertyText((BaseProperty) object.safeget(name));
+        buffer.append(XMLUtils.escapeElementText(content));
+    }
+
+    /**
+     * Get the text of the base property. Returns the empty string if the base property is null.
+     * @param property the base property
+     * @return the text of the base property. The empty string is returned if the base property is null
+     */
+    protected String getPropertyText(BaseProperty property)
```

Comment:
Missing `since` (this is a new API).

Reply:
And `@Unstable`... However I don't think it's right to make it an API. I'd keep this help method not user public (I wouldn't expose it to the world).